nekton/nii2dcm.py

```diff
@@ -0,0 +1,157 @@
+
+        return self.sort_order(z_locs, dcmfiles)
+
+    @staticmethod
+    def _create_dicomseg(
+        seg_map: Dataset, segImage: np.ndarray, dcmImage: FileDataset
+    ) -> SegmentationDataset:
+        """create a dicomseg for storage
+
+        Args:
+            seg_map (Dataset): Dataset info extraced from the mapping json
+            segImage (np.ndarray): a single slice of the segmentation nifti image
+            dcmImage (FileDataset): input dicom to which the dicomseg is to be linked to
+
+        Returns:
+            SegmentationDataset: created dicomseg file
+        """
+        # add fake storage info if necessary
+        appendedImagePosition = False
+        try:
+            dcmImage.ImagePositionPatient
+        except Exception:
+            appendedImagePosition = True
+            dcmImage.ImagePositionPatient = [0, 0, 0]
+
+        # convert to itk image
+        segImage_itk = sitk.GetImageFromArray(segImage.astype(np.uint8))
+
+        writer = pydicom_seg.MultiClassWriter(
+            template=seg_map,
+            inplane_cropping=False,  # Crop image slices to the minimum bounding box on
+            # x and y axes. Maybe not supported by other frameworks.
+            skip_empty_slices=True,  # Don't encode slices with only zeros
+            skip_missing_segment=True,  # If a segment definition is missing in the
+            # template, then raise an error instead of
+            # skipping it.
+        )
+
+        # write the image
+        dcmseg = writer.write(segImage_itk, source_images=[dcmImage])
+        # correct the acquition time and other info if neccesary
+        dcmseg.AcquisitionTime = dcmImage.AcquisitionTime
+        if appendedImagePosition:
+            dcmseg.ImagePositionPatient = None
+
+        return dcmseg
+
+    def multilabel_converter(
+        self, segfiles=List[Path], dcmfiles=List[Path]
+    ) -> List[Path]:
+        # all_individual_segs = [nib.load(file).get_fdata() for file in segfiles]
+        # de-priorisited for now
+
+        raise NotImplementedError(
+            "Multilabel NifTi s to DICOMSeg cannot to be handled yet!!"
+        )
+
+    def multiclass_converter(
+        self, segfile: Path, segMapping: Path, dcmfiles: List[Path]
+    ) -> List[Path]:
+        """Convert a given nifti segmentation to dicomseg for multiclass labels
+
+        Args:
+            segfile (Path): path to the nifti segmentation file
+            segMapping (Path): path to the dcmqii format segmentation mapping json
+            dcmfiles (List[Path]): list of paths of all the source dicom files
+
+        Returns:
+            List[Path]: list of paths of all generated dicomseg files
+        """
+        # load the segmentation mapping
+        seg_map = self._load_segmap(segMapping)
+        # load the segmentation and verify if all dicoms exist
+        seg = nib.load(segfile).get_fdata()
+        sorted_dcmfiles = self._check_all_dicoms(dcmfiles, seg)
```

Comment:
I guess here it will be important to make sure that dicoms and nifti slices are matching in terms of order and rotation.

Reply:
for the simple tests using the files they seem to match quite well. The main test will be with large nifti and dicom files if they match as well.